### DISCUSSION TITLE
Libre 2 serial numbers

### DISCUSSION THREAD

**Discussion by PhilTU** [NightscoutFoundation/xDrip#2716] [General]
Someone said to me that UK/EU Libre 2 serial numbers start with a 3. Right now my sensor is reported by xDrip+ classic status page as ABBOTT0MH00....  whereas when it was working via the patched app it was reported there as ABBOTT3MH00....

I have seen at least one post where the status page didn't match the sensor box (someone posted a screenshot and a photo together).

Is this a bug ? If the 3 does indicate a Libre 2 (?) it could be a useful diagnostic as sometimes the wrong type is issued.

**Comment by Navid200** [comment]
@tzachi-dar

**Comment by PhilTU** [comment]
I have confirmation from other users that the serrial number is misreported. Should I report this as an issue ?

**Reply by Navid200** [reply]
Please wait.  I have asked @tzachi-dar to respond.

Do you think the info is there and somehow xDrip is making a mistake?

**Reply by PhilTU** [reply]
Will wait. I saw someone refer to ABBOTT0MH as a prefix and the rest as the serial number. Perhaps this used to be the case and is reflected in the code?

xDrip is reading the serial number and getting the latter part correct so yes I think it is a mistake.

**Comment by gui-dos** [comment]


**Reply by PhilTU** [reply]
@gui-dos thanks for the clarification.

**Reply by Navid200** [reply]
OK
@PhilTU Please click on "Create issue from discussion" in the right pane.

**Reply by tzachi-dar** [reply]
Was an issue created for this?

**Reply by Navid200** [reply]
Yes: https://github.com/NightscoutFoundation/xDrip/issues/2725

**Comment by tzachi-dar** [comment]
Does this has any impact except the report of a wrong sensor value?

**Reply by PhilTU** [reply]
The only consequence is an incorrect value. This may lead one to think that a Libre 1 sensor is in use rather than a Libre 2, which would affect troubleshooting or reporting a faulty sensor to the vendor.

**Comment by tzachi-dar** [comment]
@PhilTU do you have the tag getid for such a sensor?
We can either get it for your DB if that sensor is not valid or by looking at the logs if you still have such a sensor handy.

**Reply by PhilTU** [reply]
I have the phone with the relevant sensor number still showing in xDrip,
but the sensor is no longer available. It expired over a week ago but I'm
happy to follow instructions to retrieve data.

**Comment by tzachi-dar** [comment]
So, the simplest way to get the data is if you can export your xdrip db and send it tzachi.dar@gmail.com
If that does not work, than I can explain how you can get only the specific UUID from the DB.

**Reply by PhilTU** [reply]
sent

**Comment by tzachi-dar** [comment]
Thanks, I think that I have all the info that I need, I'll report in a day or two.

**Comment by tzachi-dar** [comment]
After some digging into the issue, it seems that the difference is coming from (family.rawValue)
return fiveBitsArray.reduce("\(family.rawValue)", {
        $0 + lookupTable[ Int(0x1F & $1) ]
    })

So, the two functions are the same. It is just that somebody has decided to add the family.rawValue to the serial number. I'll have a dipper dive into the swift code and see how they are doing this.

**Comment by tzachi-dar** [comment]
@gui-dos if I understand correctly the logic is simple: if this is a libre2 sensor simply add a 3 before the serial number.
Is this correct?

**Reply by PhilTU** [reply]
![DSC_1863](https://user-images.githubusercontent.com/76240903/226169552-68df4f8d-643e-4f0b-b5eb-ba93346b9afc.JPG)
> @gui-dos if I understand correctly the logic is simple: if this is a libre2 sensor simply add a 3 before the serial number. Is this correct?

As a user I would say the 3 is the first character of the serial number. It's printed on the sensor and the box.

**Reply by filipedanunes** [reply]
@PhilTU any conclusions regarding the serial? In fact librelink reports a 3 and xdrip a O

ABBOTTOMH0165AJY4 vs  3MH0165AJY4

**Reply by PhilTU** [reply]
The issue remains open and unsolved.

**Comment by tjcbento** [comment]
Hey guys,
Not sure if this is related but I have not been able to use my past 2 sensors with Xdrip. Have been using Xdrip without any issues for at least 2 years now.

I get an NFC read timeout error and this is what I get on the logs (have already reinstalled the app around 3 times to no avail):

![31b840f4-f8be-4213-b283-d16a7388e644](https://github.com/NightscoutFoundation/xDrip/assets/134380253/e1679dd1-06a4-4c8f-b4db-92faa86ed744)


Think this will get patched soon?

**Reply by PhilTU** [reply]
what sensor do you have, and where are you ?

**Reply by tjcbento** [reply]
I threw the box away but the sensor's ID is: 3MH00NQ6YJM.
I am in Portugal.
Thanks for the quick reply :).

**Reply by PhilTU** [reply]
Ah, so this is an EU Libre 2 ?

Above it says "Sensor type unknown" and then assumes it's a Libre 1. I would check you have OOP2 version from June 22 and maybe update xDrip+ from https://jamorham.github.io/#xdrip-plus 20230503 is latest stable

**Reply by tjcbento** [reply]
Sadly, same issue:
![4953a911-5e32-4bf9-9b6d-510391ff8237](https://github.com/NightscoutFoundation/xDrip/assets/134380253/f24626df-a1c6-48b8-9d92-31deeb0ee702)

OOP2 is the latest (1.2) available from: https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip

**Reply by Navid200** [reply]
I believe @tzachi-dar is working on a PR to fix this.

**Reply by tjcbento** [reply]
Thank you for the quick feedback! For now I am using Juggluco, but it would be great to go back to Xdrip.
I appreciate the effort; this app has changed my whole life.

**Reply by parapenT1sta** [reply]
> Sadly, same issue: ![4953a911-5e32-4bf9-9b6d-510391ff8237](https://user-images.githubusercontent.com/134380253/240239731-f24626df-a1c6-48b8-9d92-31deeb0ee702.jpg)
> 
> OOP2 is the latest (1.2) available from: https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip

No, that is not the latest OOP2 version!

https://xdrip.readthedocs.io/en/latest/use/OOP/

![image](https://github.com/NightscoutFoundation/xDrip/assets/608650/9ca304b2-e5e5-41c7-81a6-7cd1a5c04828)

**Reply by PhilTU** [reply]
The later one adds functions related to WearOS only as I understand it. It isn't v1.3.

**Reply by parapenT1sta** [reply]
> The later one adds functions related to WearOS only as I understand it. It isn't v1.3.

As I can compare by the info on the website beside the wearOS option. Libre 14 days US "old" vs "new" but nothing related to the Libre EU new version.

1.2: Libre 14 US (old)
OOP2_e6 : Libre 14 US (New)

**Reply by PhilTU** [reply]
Good spot, worth a try then at least.

**Reply by parapenT1sta** [reply]
> Good spot, worth a try then at least.

Btw, a newer version of OOP2 was released: https://github.com/NightscoutFoundation/xDrip/discussions/2845#discussioncomment-5995246

**Comment by ANDREAPIPPI** [comment]
When I started using Libre 2 14 Days EU with Direct link to xDrip+, I used OOP2 1.2, but I had many "signal lost" advices each day.
Two months ago I changed OOP2 1.2 to OOP2_more_en.apk, and things now are almost perfect, I lose connection only if I'm farther than about 3m from my phone, and this obviously because Libre 2 BT signal is weaker that the MiaoMiao's signal I used with Libre 1.
I suggest to give a try to the new OOP2 release, if you haven't done already....
